pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -484,11 +521,28 @@ def __call__(
 
         else:  # We are not on the root process so go into waiting loop:
             self._waitLoop()
+            cw = iw = rw = xs = hs = pi = None
 
         # Communication solution and return
         commSol = self._communicateSolution(sol)
-
-        return commSol
+        cw = self.optProb.comm.bcast(cw, root=0)
+        iw = self.optProb.comm.bcast(iw, root=0)
+        rw = self.optProb.comm.bcast(rw, root=0)
+        xs = self.optProb.comm.bcast(xs, root=0)
+        hs = self.optProb.comm.bcast(hs, root=0)
+        pi = self.optProb.comm.bcast(pi, root=0)
```

Comment:
Could you just `bcast` `restartDict` instead of all of these?

Reply:
Absolutely, I'll make the change. You can see how old the code here is --- this is before I really understood how MPI works.